NSS-Spontaneous-Spoonbills/Sprint2
# JS - Created many to many field on employee & fixed url pattern paths

## Link to Ticket

Autolink format: hashtag and number, ex: #1094

Closes #
Doesn't close a ticket

## Description of Proposed Changes
1. Ticket refactors models to include a many to many on the employee model
1. Deleted employee_training joining table model
1. Added forward slash to bangazon2 urls and removed forward slashes on WorkforceManagement urls

##### New Employee Model
```
class Employee(models.Model):
    """
    Model for Employee Table contains department_id, computer_id, is_supervisor, name, & start_date
    """
    training = models.ManyToManyField('Training_Prog')
    department = models.ForeignKey('Department', on_delete=models.CASCADE)
    computer = models.ForeignKey('Computer', on_delete=models.CASCADE)
    is_supervisor = models.BooleanField(default=False)
    name = models.CharField(max_length=50)
    start_date = models.DateField(auto_now_add=True)

    def __str__(self):

        return f'{self.name}'
```
## Steps to Test
##### Outline the steps to test
1. Make sure you're on master
1. Fetch and check out to branch
1. Run virtual machine
1. Run any migrations you need
1. Run server
```
Note: No distinct visible changes are made but migrations have changed.
```

```
git fetch --all
git checkout js_refact_many
```

## Impacted Areas in Application

##### Modified

	WorkforceManagement/models/Department_Model.py
	WorkforceManagement/models/Employee_Model.py
	WorkforceManagement/models/Training_Model.py
	WorkforceManagement/urls.py
	bangazon2/urls.py

##### Deleted

	WorkforceManagement/models/Emp_Training_Model.py

##### Added 

	WorkforceManagement/migrations/0002_employee_training.py


## Mentions @username
@binauralbeat @obscure-shadow @CashewRose @MrErin 
Tag users that need to review this code

## Testing
No testing was done on this application



## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
1. Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
